Quadriga/src/main/java/edu/asu/spring/quadriga/rest/ConceptCollectionRestController.java

```diff
@@ -448,5 +519,58 @@
         }
 
     }
+    
+    
+    /**
+     * Rest interface to get conceptcollections related to project
+     * http://<<URL>:<PORT>>/quadriga/auth/rest/<projectid>/conceptcollections.json
+     * http://localhost:8080/quadriga/auth/rest/PROJ_bb7ad41b-3e85-4309-b2ff-47d644307b9b/conceptcollections.json
+     * 
+     * 
+     * @author Ajay Modi & Bharath Srikantan
+     * @param projectid
+     * @param model
+     * @param principal
+     * @return
+     */
 
+    @RequestMapping(value = "auth/rest/{projectid}/conceptcollections.json", method = RequestMethod.GET, produces = "application/json")
+    public ResponseEntity<String> listProjectConceptCollectionJson(
+            @PathVariable("projectid") String projectid, Model model,
+            Principal principal) {
+        String userId = principal.getName();
+        List<IProjectConceptCollection> projectConceptCollectionList = null;
+        try {
+            // TODO: listProjectConceptCollection() needs to be modified
+            projectConceptCollectionList = projectConceptCollectionManager
+                    .listProjectConceptCollection(projectid, userId);
+            
+            JSONArray ja = new JSONArray();
+            
+            if(projectConceptCollectionList!=null){
+                for (IProjectConceptCollection conceptCollection : projectConceptCollectionList) {
+                    JSONObject j = new JSONObject();
+                    try {
+
+                        j.put("id", conceptCollection.getConceptCollection()
+                                .getConceptCollectionId());
+                        j.put("name", conceptCollection.getConceptCollection()
+                                .getConceptCollectionName());
+                        ja.put(j);
+
+                    } catch (JSONException e) {
+                        logger.error("JSONException:", e);
+                        return new ResponseEntity<String>(e.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+                    }
+
+                }
+            }
+            
+            return new ResponseEntity<String>(ja.toString(), HttpStatus.OK);
+            
+        } catch (QuadrigaStorageException e) {

```

Comment:
you mean: 
```
j.put("id", conceptCollection.getConceptCollection() .getConceptCollectionId());
j.put("name", conceptCollection.getConceptCollection().getConceptCollectionName());
```
? 
Yes, the json exception is fine where it is, but the Quadriga exception should be thrown before.